packages/jira-adapter/src/filters/fields/account_id_filter.ts

```diff
@@ -0,0 +1,176 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Element, getChangeData, isInstanceElement } from '@salto-io/adapter-api'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import { transformValues, TransformFunc } from '@salto-io/adapter-utils'
+import { collections } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import JiraClient from '../../client/client'
+import { FilterCreator } from '../../filter'
+
+const { createPaginator, getWithItemIndexPagination } = clientUtils
+const { makeArray } = collections.array
+const { awu, toArrayAsync } = collections.asynciterable
+export const accountIdsFieldNames = ['leadAccountId', 'authorAccountId', 'accountId']
+const accountIdString = 'ACCOUNT_ID'
+type IdDictionary = {[id: string]: string}
+type IdDictionaryStruct = {
+  dictionary: IdDictionary
+  shouldUse: boolean
+  initialized: boolean
+  client?: JiraClient
+}
+
+const CreateNewIdDictionary = async (client: JiraClient)
+: Promise<IdDictionary> => {
+  const newIdDictionary: IdDictionary = {}
+  const myPaginator = createPaginator({
+    client,
+    paginationFuncCreator: () => getWithItemIndexPagination(0, 50),
+  })
+  const paginationArgs = {
+    url: '/rest/api/3/users/search',
+    paginationField: 'startAt',
+  };
+  (await toArrayAsync(myPaginator(
+    paginationArgs,
+    page => makeArray(page) as clientUtils.ResponseValue[]
+  // eslint-disable-next-line array-callback-return
+  ))).flat().map(x => {
+    newIdDictionary[x.accountId as string] = x.displayName as string
+  })
+  return newIdDictionary
+}
+
+const getIdDictionary = async (dictionaryStruct: IdDictionaryStruct)
+: Promise<IdDictionary> => {
+  if (!dictionaryStruct.initialized
+    && dictionaryStruct.client !== undefined) {
+    dictionaryStruct.dictionary = await CreateNewIdDictionary(dictionaryStruct.client)
+  }
+  return dictionaryStruct.dictionary
+}
+
+const enhanceAccountId = async (idDictionaryStruct : IdDictionaryStruct):
+  Promise<TransformFunc> => async ({ value }) => {
+  if (_.isPlainObject(value)) {
+    await awu(accountIdsFieldNames).forEach(async fieldName => {
+      if (Object.prototype.hasOwnProperty.call(value, fieldName)) {
+        value[fieldName] = {
+          accountIdData: value[fieldName],
+        }
+        if (idDictionaryStruct.shouldUse
+          && Object.prototype.hasOwnProperty.call(
+            await getIdDictionary(idDictionaryStruct),
+            value[fieldName].accountIdData,
+          )) {
+          value[fieldName].displayName = (await
+          getIdDictionary(idDictionaryStruct))[value[fieldName].accountIdData]
```

Comment:
I think it will be more readable of you separate it to different var
```suggestion
          const idToDisplayName = await getIdDictionary(idDictionaryStruct)
          value[fieldName].displayName = idToDisplayName[value[fieldName].accountIdData]
```